### PR TITLE
chore: Remove usage tracking

### PR DESCRIPTION
**Pull request by tokoko**
# What this PR does / why we need it:
This PR removes usage tracking code entirely from feast codebase and tests. We already made it as an opt-in feature previously, which probably no one is going to opt in for :)

### CODE REVIEW

File: sdk/python/feast/batch_feature_view.py [hunk 1]
```diff
@@ -60,7 +60,7 @@ def __init__(
         *,
         name: str,
         source: DataSource,
-        entities: Optional[Union[List[Entity], List[str]]] = None,
```
Comment (franciscojavierarceo) [inline sdk/python/feast/batch_feature_view.py:63]: This seems to be u related to usage fyi

Comment (tokoko) [reply sdk/python/feast/batch_feature_view.py:63]: I know, after I removed decorators, a lot of linter rules started to fail and I had to fix a bunch of them (~20). Probably something with mypy and decorators not getting along, not sure why it wasn't an issue before.

Comment (franciscojavierarceo) [reply sdk/python/feast/batch_feature_view.py:63]: Nice, can you add that to the description and PR notes?

**Review by franciscojavierarceo — APPROVED** [review]
Nice!

**Pull request merged**